### PR TITLE
CLI-322: Add '-' as an option char and implemented extensive tests

### PR DESCRIPTION
**Pull request by Claudenw**
Fix for CLI-322

Adds kabab-format options and extensive option name testing.

### CODE REVIEW

**Comment by garydgregory** [comment]
@Claudenw 
I think you mean "kabab" -> "kebab". Always run `mvn` before you push with the latest from git master.

**Comment by Claudenw** [comment]
> @Claudenw I think you mean "kabab" -> "kebab". Always run `mvn` before you push with the latest from git master.

Indeed, I do not know how to spell. :(

I did run 'mvn' before the push and everything worked.  Got "BUILD SUCCESS" result.  Just tried again and same thing.  I am addressing the issues from the build now.

**Comment by Claudenw** [comment]
The build fails on 22-ea which fails with a spotbugs execution error,  It seems like spotbugs is not yet supported on 22-ea.

**Comment by garydgregory** [comment]
RE 22-ea, that should be fixed by the next commons-parent release, so don't worry about that part.

File: src/main/java/org/apache/commons/cli/OptionValidator.java [hunk 1]
```diff
@@ -23,14 +23,35 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
+    /** The array of additional characters allowed in the rest of the option but not in the first position */
+    static final char[] ADDITIONAL_LONG_CHARS = {'-'};
+
+    /** 
+     * Checks the char array for a matching char.
+     * @param ary the array to search
+     * @param c the char to look for.
+     * @return {@code true} if {@code c} was in {@code ary}, {@code false} otherwise. 
+     */
+    private static boolean search(final char[] ary, final char c) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/cli/OptionValidator.java:38]: 'ary' is a poor parameter name here, almost any else else is better: 'chars', 'array, 'buffer'...

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @Claudenw 
The site needs matching updates to document this feature.

**Comment by Claudenw** [comment]
@garydgregory updated as requested

**Comment by garydgregory** [comment]
Hi @Claudenw 
Run `mvn` locally before you push to catch these build failures.

**Pull request merged**